src/io/tests/test_hdf_manual_reorder.cpp

```diff
@@ -0,0 +1,210 @@
+//
+// File created by: Luke Shulenburger, lshulen@sandia.gov, Sandia National Laboratories
+//////////////////////////////////////////////////////////////////////////////////////
+
+
+#include "catch.hpp"
+#include "io/hdf_archive.h"
+#include <iostream>
+#include <vector>
+#include <OhmmsPETE/OhmmsVector.h>
+#include <OhmmsPETE/OhmmsMatrix.h>
+#include <OhmmsPETE/OhmmsArray.h>
+
+using std::cout;
+using std::endl;
+using std::vector;
+
+using namespace qmcplusplus;
+
+
+/*
+TEST_CASE("hdf_write_reorder", "[hdf]")
+{
+  hdf_archive hd;
+  hd.create("test_write_reorder.hdf");
+
+  vector<double> v{1.1, -2.2, 3.3, -4.4, 5.5, -6.6, 7.7, -8.8, 9.9};
+  vector<hsize_t> dims{3,3};
+  bool okay = hd.writeEntry(v, dims, "array_from_vector");
+  REQUIRE(okay);  
+  hd.close();
+
+  hdf_archive hd2;
+  okay = hd2.open("test_write_reorder.hdf");
+  REQUIRE(okay);
+
+  Tensor<double,3> read_tensor;
+  hd2.read(read_tensor, "array_from_vector");
+  REQUIRE(read_tensor.size() == v.size());
+  for (int i = 0; i < 3; i++)
+  {
+    for (int j = 0; j < 3; j++)
+    {
+      int pos = i*3+j;
+      REQUIRE(v[pos] == read_tensor(i,j));
+    }
+  }
+  hd2.close();
+}
+*/
+
+TEST_CASE("hdf_write_reorder_with_matrix", "[hdf]")
+{
+  hdf_archive hd;
+  bool okay = hd.create("test_write_matrix_reorder.hdf");
+  REQUIRE(okay);
+  
+  Vector<double> v(6);
+  v[0] = 0.0;
+  v[1] = 0.1;
+  v[2] = 0.2;
+  v[3] = 1.0;
+  v[4] = 1.1;
+  v[5] = 1.2;
+
+  Matrix<double> matrix_view(v.data(), 2, 3);
+  hd.write(matrix_view, "matrix_from_vector");
+  
+  hdf_archive hd2;
+  hd2.open("test_write_matrix_reorder.hdf");
+  
+  Matrix<double> m(2,3);
+  hd2.read(m, "matrix_from_vector");
+
+  REQUIRE(m(0,0) == Approx(v[0]));
+  REQUIRE(m(0,1) == Approx(v[1]));
+  REQUIRE(m(0,2) == Approx(v[2]));
+  REQUIRE(m(1,0) == Approx(v[3]));
+  REQUIRE(m(1,1) == Approx(v[4]));
+  REQUIRE(m(1,2) == Approx(v[5]));
+}
+
+
+// do this in three ways.  
+// 1. relying on my overloading in std::vector (do not perfer this, too specific)
+// 2. rely on my understanding of hyperslab_proxy (seems to require you to know
+//          too much about the dimensions in the dataset and is prone to user error)
+// 3. use hyperslab_proxy to handle the back end api, but specify the slice to choose
+//          with a vector and query the dataset to figure out the other needed
+//          parts.  better than #1 because it does not specially call out vector
+TEST_CASE("hdf_read_partial", "[hdf]")
+{
+  hdf_archive hd;
+  bool okay = hd.create("test_read_partial.hdf");
+  REQUIRE(okay);
+  
+  Matrix<double> allData(3,4);
+  for (int i = 0; i < 3; i++) {
+    for (int j = 0; j < 4; j++) {
+      allData(i,j) = i+j*0.1;
+    }
+  }
+
+  hd.write(allData, "matrix");
+  hd.close();
+
+  hdf_archive hd2;
+  okay = hd2.open("test_read_partial.hdf");
+  REQUIRE(okay);
+
+  // method 1 (relying on vector in hdf_stl)
+  vector<int> readSpec{1,-1};
```

Comment:
naming conventions. And here (std::)vector isued while ohhms Vector was used above.

Reply:
The initial code I wrote was using std::vector for the descriptor.  In method 3, I switched to a templated integer container type (IC) as is used elsewhere in the hdf_archive stuff.  That way one could use either.

Reply:
That's probably better then.